src/unistd.rs

```diff
@@ -1326,6 +1327,45 @@ pub fn pause() -> Result<()> {
     Errno::result(res).map(drop)
 }
 
+/// Schedule an alarm signal.
+///
+/// Returns a previously set alarm, if any.
+///
+/// See also [alarm](http://pubs.opengroup.org/onlinepubs/9699919799/functions/alarm.html).
+///
+/// # Note
+///
+/// Alarm only allows for times with a second granularity, anything smaller then
+/// a second will be ignored (not rounded up). E.g. 1 second and 800 miliseconds
+/// will become 1 second.
+///
+/// Also, Strictly conforming the POSIX spec, duriations longer then `65535`
```

Comment:
IMHO the warning in the POSIX specification is dumb.  It just warns you against using constants greater than `UINT_MAX`, which is something your compiler can already warn you about.  However, this Rust wrapper introduces a new problem: `secs` is a `u64`, which can overflow at runtime.  You should probably `assert` that `secs <= libc::c_uint::max_value()`

Reply:
You want just an assert, or a silent convertion into the max value with a warning in the doc?

Reply:
I think an assert would be better than a saturating conversion.  After all, why would somebody ever set an alarm for more than 4 billion seconds into the future?  Such a thing is probably a bug.